DOCS-9559-DW-datetime-example-kt (#1736)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-types-coercion.adoc

                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
----

> The following example converts two strings into date format and concatenates the result. The first string represents the date, the second string represents the time. The transformation uses `as` to coerce the first string to `LocalDateTime`, and then to a `String` with the desired format. The transformation also uses `++` to concatenate the result.

Reviewer: ```suggestion
The following example converts two strings into a date format and concatenates the result. The first string represents the date, and the second string represents the time. The transformation uses `as` to coerce the first string to `LocalDateTime`, and then to a `String` with the specified format. The transformation also uses `++` to concatenate the result.
```

.DataWeave script:
[source,dataweave,linenums]
----
%dw 2.0
output application/json
var s1= 20201228 // (yyyymmdd),
var s2= 1608 //(hhmm)
---
(s1 ++ s2)

> as LocalDateTime {format:"yyyyMMddHHmm"}

Reviewer: We recommend using `uuuu` for dates because `yyyy` is for dates that involve A.D. (ACE)  and B.C. (BCE) eras. See https://docs.mulesoft.com/mule-runtime/4.3/dataweave-cookbook-format-dates#ex03
specifically, `//y is for use with the era (BCE or CE). Generally, use u, instead.
    "year-y" : myDateTime as String { format: "y"},`

as String {format:"MM-dd-yyyy HH:mm:ss"}
----

.Output:
[source,json,linenums]
----
"12-28-2020 16:08:00"
----

> The following example is similar to the previous one and converts a string into date format. First the transformation uses `as` to coerce the string into `LocalDateTime`, and then coerces it again to `String` with the desired output format.

Reviewer: ```suggestion
The following example also converts a string into a date format. First, the transformation uses `as` to coerce the string into a `LocalDateTime` type, and then it coerces the result into a `String` type with the specified output format.
```
